pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/AndDocIdIterator.java

```diff
@@ -18,112 +18,52 @@
  */
 package org.apache.pinot.core.operator.dociditerators;
 
-import java.util.Arrays;
 import org.apache.pinot.core.common.BlockDocIdIterator;
 import org.apache.pinot.core.common.Constants;
 
 
-// TODO: Optimize this
 public final class AndDocIdIterator implements BlockDocIdIterator {
```

Comment:
1.javadoc?
2. comments about thread safety. is it safe/advisable to use this iterator in more than one thread?

Reply:
It is not thread safety (the whole filtering layer does not involve thread safety because it is happening under one segment which is always processed by one thread). Also, I don't think multiple threads using one iterator make sense..